superset/db_engine_specs/base.py

```diff
@@ -1250,6 +1249,22 @@ def get_extra_params(database: "Database") -> Dict[str, Any]:
                 raise ex
         return extra
 
+    @staticmethod
+    def get_encrypted_extra_params(database: "Database") -> Dict[str, Any]:
```

Comment:
Why moving this logic from Database model to here? At first I thought you'd be overriding it for Trino but then not, so I'm not clear why moving the logic to `BaseEngineSpec`?

Reply:
Actually, YES. But it's my organization specific code, so I not put it here.
But I still push the change of `get_encrypted_extra_params` in order to align with `get_extra_params`, and I think many people out there also need this.

Reply:
Hello @mistercrunch, I just move this change to another PR.
And the rest, could you please take a look.